l10n_ch_qr_bill/models/res_bank.py

```diff
@@ -24,6 +24,18 @@ class ResPartnerBank(models.Model):
         )
     )
 
+    @api.multi
+    @api.constrains('l10n_ch_qr_iban')
+    def _check_ch_li_l10n_ch_qr_iban(self):
+        """Check if field is populated with Swiss / Liechtenstein qr iban"""
+        for record in self:
+            if (
+                record.l10n_ch_qr_iban
+                and not record.l10n_ch_qr_iban.startswith(('CH', 'LI'))
+            ):
+                raise ValidationError(_(
+                    "Not valid Switzerland or Liechtenstein QR IBAN."))
```

Comment:
Please use `_validate_qr_iban` instead.

And move the CH LI check in there.

It seems `_validate_qr_iban` is there for that, the constraint was missing though.

```suggestion
            if record.l10n_ch_qr_iban:
                self._validate_qr_iban(record.l10n_ch_qr_iban)
```

Reply:
Done